contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -5743,33 +5744,64 @@ static List *transformSelectIntoStmt(CreateTableAsStmt *stmt, const char *queryS
 					}
 				}
 
-				into->identityName = tle->resname;
-				into->identityType = TypeNameToString(typename);
 				seen_identity = true;
+				identity_ressortgroupref = tle->ressortgroupref; /** Save this Index to modify sortClause and distinctClause*/
+
+				/** Add alter table add identity node after Select Into statement */
+				altstmt = makeNode(AlterTableStmt);
+				altstmt->relation = into->rel;
+				altstmt->cmds = NIL;
+
+				constraint = makeNode(Constraint);
+				constraint->contype = CONSTR_IDENTITY;
+				constraint->generated_when = ATTRIBUTE_IDENTITY_ALWAYS;
+				constraint->options = seqoptions;
+
+				def = makeNode(ColumnDef);
+				def->colname = tle->resname;
+				def->typeName = typename;
+				def->identity = ATTRIBUTE_IDENTITY_ALWAYS;
+				def->is_not_null = true;
+				def->constraints = lappend(def->constraints, constraint);
+
+				lcmd = makeNode(AlterTableCmd);
+				lcmd->subtype = AT_AddColumn;
+				lcmd->missing_ok = false;
+				lcmd->def = (Node *)def;
+				altstmt->cmds = lappend(altstmt->cmds, lcmd);
+			}
+			else
+			{
+				current_resno += 1;
+				tle->resno = current_resno;
+				modifiedTargetList = lappend(modifiedTargetList, tle);
+			}
+		}
+		q->targetList = modifiedTargetList;
 
-				snamespaceid = RangeVarGetCreationNamespace(into->rel);
-				snamespace = get_namespace_name(snamespaceid);
-				sname = ChooseRelationName(into->rel->relname, tle->resname, "seq", snamespaceid, false);
-				seqstmt = makeNode(CreateSeqStmt);
-				seqstmt->for_identity = true;
-				seqstmt->sequence = makeRangeVar(snamespace, sname, -1);
-				seqstmt->sequence->relpersistence = into->rel->relpersistence;
-				seqstmt->options = seqoptions;
-
-				altseqstmt = makeNode(AlterSeqStmt);
-				altseqstmt->sequence = makeRangeVar(snamespace, sname, -1);
-				attnamelist = list_make3(makeString(snamespace), makeString(into->rel->relname), makeString(tle->resname));
-				altseqstmt->options = list_make1(makeDefElem("owned_by", (Node *)attnamelist, -1));
-				altseqstmt->for_identity = true;
+		if (seen_identity)
+		{
+			if (q->sortClause)
+			{
+				List *modifiedSortList = NIL;
+				ListCell *olitem;
+				foreach (olitem, q->sortClause)
+				{
+					SortGroupClause *sortcl = (SortGroupClause *)lfirst(olitem);
```

Comment:
Always check IsA() to make sure we are getting the correct struct

Reply:
Done